### PR TITLE
Driver/keithley2600 fast sweep

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
Relates to #720.

Changes proposed in this pull request:
- Add a channelised driver for the Keithley2600
- Make the channelised driver able to do fast sweeps using Lua scripts
- Add a notebook benchmarking the performance

@jenshnielsen @Dominik-Vogel

### CODE REVIEW

File: qcodes/instrument_drivers/tektronix/Keithley_2600.py [hunk 1]
```diff
@@ -1,22 +1,101 @@
...
         TODO:
-        - Add all parameters that are in the manual
-        - range and limit should be set according to mode
+        - Make a channelised version for the two channels
         - add ramping and such stuff
 
     """
-    def __init__(self, name, address, channel, **kwargs):
+    def __init__(self, name: str, address: str, channel: str,
+                 model: str=None, **kwargs) -> None:
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600.py]: I think the typo for model should be `Optional[str]`

Comment (jenshnielsen) [reply qcodes/instrument_drivers/tektronix/Keithley_2600.py]: or maybe not?

File: qcodes/instrument_drivers/tektronix/Keithley_2600.py [hunk 2]
```diff
@@ -1,22 +1,101 @@
...
+            model: The model type, e.g. '2614B'
+        """
+
+        warnings.warn("This Keithley driver is old and will be removed "
+                      "from QCoDeS soon. Use Keithley_2600_channels "
+                      "instead, it is MUCH better.", UserWarning)
+
         super().__init__(name, address, terminator='\n', **kwargs)
+
+        if model is None:
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600.py]: Could we do this smarter. Is there a way to get the model from the api

File: qcodes/instrument_drivers/tektronix/Keithley_2600.py [hunk 3]
```diff
@@ -35,26 +114,51 @@ def __init__(self, name, address, channel, **kwargs):
...
-        # needs get after set
-        self.add_parameter('rangev',
+        self.add_parameter('nplc',
+                           label='Number of power line cycles',
+                           set_cmd='measure.nplc={:.4f}',
+                           get_cmd='measure.nplc',
+                           get_parser=float,
+                           vals=vals.Numbers(0.001, 25))
+        # volt range
+        # needs get after set (WilliamHPNielsen): why?
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600.py]: in which way. To be up to date?

File: qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+
+class Keithley_2600(VisaInstrument):
+    """
+    This is the qcodes driver for the Keithley_2600 Source-Meter series,
+    tested with Keithley_2614B
+
+    Status: beta-version.
+        TODO:
+        - Make a channelised version for the two channels
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py]: remove this todo?

File: qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+                                   1e-3, 10e-6, 100e-3, 1, 1.5]}
+
+        # Add the channel to the instrument
+        channels = ChannelList(self, "Channels", KeithleyChannel,
+                               snapshotable=False)
+        for ch in ['a', 'b']:
+            ch_name = 'smu{}'.format(ch)
+            channel = KeithleyChannel(self, ch_name, ch_name)
+            channels.append(channel)
+            self.add_submodule(ch_name, channel)
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py]: personally I would leave this out and make the channellist snapshotable. 
As this is you can do

```
key.smua.v
key.channels.smua.v
key.channels[0].v
```
I think it's more logical to only have access via channels

File: qcodes/instrument_drivers/tektronix/Keithley_2600.py (hunk 3 above)
Comment (WilliamHPNielsen) [reply qcodes/instrument_drivers/tektronix/Keithley_2600.py]: I only added the "why", the comment was already in the driver.

Comment (jenshnielsen) [reply qcodes/instrument_drivers/tektronix/Keithley_2600.py]: Fair enough I guess I'm not the only one who thinks this is unclear

**Comment by WilliamHPNielsen** [comment]
@jenshnielsen, I think this is ready to 💃

**Pull request merged**